group/views.py

```diff
@@ -195,3 +253,18 @@ def edit(request, group_id):
                     return HttpResponseRedirect('/group/detail/%s' % modified_group.id)
         else:
             raise PermissionDenied
+
+def add(request, group_id):
+    group = get_group(group_id)
+
+    if has_group_ownership(request.user, group):
+        if request.method == 'POST':
+            form = AnnounceForm(request.POST)
+            if form.is_valid():
+                new_announce = form.save()
```

Comment:
If none of the condition is satisfied, it will return ``None``
Maybe
```py
if has_group_ownership(request.user, group) and request.method == 'POST' and form.is_valid():
    # code 
else:
    raise PermissionDenied
```